table/block_based/index_builder.h

```diff
@@ -187,6 +189,12 @@ class IndexBuilder {
                      l_user_key, r_user_key) == 0;
   }
 
+  // Updates the cached index size estimate used by CurrentIndexSizeEstimate().
+  // Subclasses that keep a cached estimate should override this method
+  // and call it whenever their internal state changes (such as when index
+  // entries are added).
+  virtual void UpdateIndexSizeEstimate() {}
```

Comment:
This function can be called from the SST "write thread" (via FinishIndexEntry()), and needs to be thread safe with CurrentIndexSizeEstimate() called from the SST "emit thread".

I believe that is implemented properly here, just needs to be noted in comments